server/ladders.ts

```diff
@@ -163,7 +163,12 @@ class Ladder extends LadderStore {
 			connection.popup(`You are already challenging someone. Cancel that challenge before challenging someone else.`);
 			return false;
 		}
-		if (targetUser.settings.blockChallenges && !user.can('bypassblocks', targetUser)) {
+		if (
+			user.settings.blockChallenges &&
+			(user.settings.blockChallenges === true || !Users.globalAuth.atLeast(targetUser, user.settings.blockChallenges)) &&
+			!targetUser.can('bypassblocks') &&
+			targetUser.id !== user.id
```

Comment:
```suggestion
			!targetUser.can('bypassblocks')
```
Users can't challenge themselves, anyway.